### PR TITLE
3580 Document Viewer

### PR DESCRIPTION
**Pull request by matthopson**
Screen Shot 2020-07-02 at 11 30 36 AM

### CODE REVIEW

File: web-client/src/presenter/actions/getDefaultDocketViewerDocumentToDisplayAction.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * gets the first docket entry document from the current case detail to set as the default viewerDocumentToDisplay
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.get the cerebral get method
+ * @returns {object} object containing viewerDocumentToDisplay
+ */
+export const getDefaultDocketViewerDocumentToDisplayAction = ({ get }) => {
+  let viewerDocumentToDisplay = null;
+  const { formattedDocketEntries } = get(state.formattedCaseDetail);
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/getDefaultDocketViewerDocumentToDisplayAction.js]: Is it ok to use a computed value in an action? Do we do this in other places?

Comment (matthopson) [reply web-client/src/presenter/actions/getDefaultDocketViewerDocumentToDisplayAction.js]: Good call. Lemme refactor that

**Review by rachaelparris — APPROVED** [review]

File: web-client/src/presenter/actions/setViewerDocumentToDisplayAction.js [hunk 2]
```diff
@@ -1,32 +1,32 @@
 import { state } from 'cerebral';
 
 /**
- * sets the attachmentDocumentToDisplay from props
+ * sets the viewerDocumentToDisplay from props
  *
  * @param {object} providers the providers object
```
Comment (sutt0n) [inline web-client/src/presenter/actions/setViewerDocumentToDisplayAction.js:6]: nit: missing jsdoc for `get`

File: web-client/src/views/DocketRecord/DocumentViewer.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    viewerDocumentToDisplay,
+  }) {
+    return (
+      <>
+        <div className="grid-row grid-gap-5">
+          <div className="grid-col-4">
+            <div
+              className={classNames(
+                'document-viewer--documents',
+                !formattedCaseDetail.formattedDocketEntries.length &&
```
Comment (sutt0n) [inline web-client/src/views/DocketRecord/DocumentViewer.jsx]: i feel like we check the length for formattedDocketEntries in a lot of places (or just checking the length of arrays in general for view logic) -- i wonder if it'd be best to just return a bool, `hasDocketEntries: formattedCaseDetail.formattedDocketEntries.length`. thoughts?

Comment (matthopson) [reply web-client/src/views/DocketRecord/DocumentViewer.jsx]: Yeah, I wanted to add this when we get to adding helper

**Review by sutt0n — APPROVED** [review]

File: web-client/src/views/DocketRecord/DocumentViewer.jsx (hunk 3 above)
Comment (rachaelparris) [reply web-client/src/views/DocketRecord/DocumentViewer.jsx]: Is it ever possible for `formattedDocketEntries` to have length 0? There will always be a petition document, right?

Comment (sutt0n) [reply web-client/src/views/DocketRecord/DocumentViewer.jsx]: yeah, we do _quite_ a few length checks, it'd probably be worth it to make a devex task

**Review by rachaelparris — APPROVED** [review]

**Pull request merged**